File: src/main/java/nu/fgv/register/server/mapper/SpexCategoryMapper.java
package nu.fgv.register.server.mapper;

import nu.fgv.register.server.dto.SpexCategoryDto;
import nu.fgv.register.server.model.SpexCategory;
import org.mapstruct.Mapper;

import java.util.List;

@Mapper(
        componentModel = "spring"
)
public interface SpexCategoryMapper {

    SpexCategoryDto toSpexCategoryDto(SpexCategory model);

    List<SpexCategoryDto> toSpexCategoryDtos(List<SpexCategory> models);

    SpexCategory toSpexCategory(SpexCategoryDto dto);
}

File: src/main/kotlin/nu/fgv/register/server/dto/mapper/SpexCategoryMapper.kt
package nu.fgv.register.server.dto.mapper

import nu.fgv.register.server.domain.SpexCategory
import nu.fgv.register.server.dto.SpexCategoryDto
import org.mapstruct.Mapper

@Mapper(componentModel = "spring", uses = [])
interface SpexCategoryMapper : EntityMapper<SpexCategoryDto, SpexCategory> {

    fun fromId(id: Long?): SpexCategory? {
        if (id == null) {
            return null
        }
        val spexCategory = SpexCategory()
        spexCategory.id = id
        return spexCategory
    }
}
